Feature/main screen

Roller23 opened the pull request:
Key changes so far:

- Changed the font from Lato to Inter
- Added the ability to sign in by clicking enter instead of clicking the button
- Added status messages
- Added input animations
- Added fade out animation
- You're logged in automatically after logging in once (you can log off later)

![ezgif-7-981ea3da6854](https://user-images.githubusercontent.com/25384028/113785048-abb92d80-9736-11eb-9b30-e99f1481600f.gif)

![image](https://user-images.githubusercontent.com/25384028/113785227-fe92e500-9736-11eb-8a99-438e9cebef17.png)

This is a draft and probably should not be merged just yet

To do:

- implement a basic main hub

Roller23 committed:
changed font from Lato to Inter

Roller23 committed:
added different login button texts depending on the context

Roller23 committed:
moved styles to separate modules

Roller23 committed:
added input animations

adacienciala commented:
Overall looks great! Can the dots in "Authorizing..." be animated?

Roller23 commented:
@adacienciala yeah I'd like to see some kind of animation here, either these three dots changing or some spinner like

![image](https://i.stack.imgur.com/h5tnp.gif)

would be pretty cool

Roller23 commented:
OR maybe... mini rain animation

byteingWolf commented on scripts/javascript/utils.js:
```diff
@@ -48,3 +52,17 @@ function authSocket() {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+exports.sleep = sleep;
+function fadeOut(el, ms) {
```
Hardcoded time handling could create a problem in future

Roller23 replied:
Thanks for the heads up, fixed

byteingWolf reviewed (approved):
Overall looks great but please type comments where funtion are termopary and/or you are not sure about their usage in future.

Roller23 committed:
maximizing on successful login + minor changes

Roller23 committed:
additional input alerts

Roller23 committed:
added basic hub outline and messaging

Roller23 committed:
fixed sign up not showing up

Roller23 committed:
removed bad module import

Roller23 committed:
fixed messages scrolling

Roller23 committed:
scrolling down only when needed

Roller23 committed:
replaced input with textarea

Roller23 committed:
added prettier scrollbar and fixed login input margin

Roller23 commented:
Progress report

- it's now possible to send messages to a general test channel
- some fixes

<img width="1045" alt="Screenshot 2021-04-07 at 22 15 29" src="https://user-images.githubusercontent.com/25384028/113928450-1d54b280-97ef-11eb-929b-a4292695ffa9.png">

- New scrollbar look proposed by me (tbc)

<img width="320" alt="Screenshot 2021-04-07 at 22 15 58" src="https://user-images.githubusercontent.com/25384028/113928464-2180d000-97ef-11eb-8f72-b489941da22c.png">

Still no hub

Roller23 committed:
preventing double socket listeners

Roller23 committed:
a better solution to prevent multiple listeners on a single event

Roller23 committed:
moved button texts from css to js

Commit:
added window control buttons on the main screen + fixed a few bugs

Roller23 committed:
added 9 new splash screens

Roller23 committed:
two new splash screens

Roller23 committed:
updated bg1

Roller23 committed:
code refactoring + comments

Roller23 committed:
added date formatting for all timezones with moment.js

Roller23 committed:
fixed message separation

Roller23 committed:
minor changes

Roller23 committed:
added support for multiple message channels

Roller23 committed:
fixed message time bug

Roller23 committed:
added dates for submessages

Roller23 committed:
reverted tile colors

byteingWolf commented:
There is a problem with filling text boxes after inncorrect login attempt. 

https://user-images.githubusercontent.com/69678128/115070240-fab05100-9ef4-11eb-87e6-290588b3111e.mp4



Lets try correct it :)

Roller23 committed:
added a new login background

Roller23 committed:
resized the side panels

Roller23 committed:
added the settings and profile buttons

Roller23 committed:
added all left side buttons

Roller23 committed:
showing the left panel by default

Roller23 commented:
Progress update:

Implemented the new Figma layouts partially, added animations and tooltips


https://user-images.githubusercontent.com/25384028/120728161-95341600-c4dc-11eb-831b-a670a2685dfc.mov

Roller23 committed:
fixed icons alignment on the left

Roller23 committed:
added variables

Roller23 committed:
exported panel toggle to functions

Roller23 committed:
changed user's placeholder

Roller23 committed:
left bar progress

Roller23 committed:
added section for servers

Roller23 committed:
refactored tooltips code

Roller23 committed:
code refactoring for tooltips

Roller23 committed:
removed fixed string

Roller23 commented:
This PR is going for far too long, I think it's better to merge it and then split work into separate branches and PRs

szezjo reviewed (approved):
Yes.
